lib/stat_tracker.rb

```diff
@@ -113,4 +113,58 @@ def worst_coach(season)
 
     coach_results.min_by { |k, v| v }[0]
   end
+
+  def calculate_team_accuracies(season)
+    team_accuracies = {}
+
+    game_teams_in_season = @game_teams_data.select do |game_team|
+      game_id = game_team[:game_id]
+      game = @games_data.find { |game| game[:game_id] == game_id }
+    
+      if game && game[:season] == season
+        true
+      else
+        false
+      end
+    end
+
+   game_teams_in_season.each do |game_team|
+    team_id = game_team[:team_id]
+    goals = game_team[:goals].to_i
+    shots = game_team[:shots].to_i
+
+      if shots > 0
+        accuracy_ratio = goals.to_f / shots
+        team_accuracies[team_id] ||= [] 
+        team_accuracies[team_id] << accuracy_ratio
+      end
+    end
+
+    team_accuracies.each do |team_id, ratios|
+      average_accuracy = (ratios.reduce(:+) / ratios.size).round(2)
+      team_accuracies[team_id] = average_accuracy
+    end
+
+    team_accuracies
+  end
+
+  def most_accurate_team(season)
+    team_accuracies = calculate_team_accuracies(season)
```

Comment:
move `#max_by` up to line 152; rename to `highest_accuracy`

Reply:
changed method